webpack/assets/javascripts/react_app/components/ToastsList/index.js

```diff
@@ -1,9 +1,64 @@
-import { connect } from 'react-redux';
-import * as ToastActions from '../../redux/actions/toasts';
-import ToastsList from './ToastList';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
 
-const mapStateToProps = state => ({
-  messages: state.toasts.messages,
-});
+import {
+  Alert,
+  AlertActionCloseButton,
+  AlertActionLink,
+  AlertGroup,
+} from '@patternfly/react-core';
 
-export default connect(mapStateToProps, ToastActions)(ToastsList);
+import { addToast, deleteToast } from '../../redux/actions/toasts';
+import { selectToastsListMessages } from './ToastsListSelectors';
+import { toastType, toastTitle } from './ToastListHelpers';
+import './ToastList.scss';
+
+const ToastsList = ({ railsMessages }) => {
+  const dispatch = useDispatch();
+  const messages = useSelector(selectToastsListMessages);
+
+  useEffect(() => {
+    railsMessages.forEach(({ message, type, key }) => {
+      dispatch(addToast({ message, type, key }));
+    });
+  }, [dispatch, railsMessages]);
+
+  const toastsList = Object.entries(messages)
+    .map(([key, message]) => ({ key, ...message }))
+    .map(({ key, type, title, message, link, sticky, ...toastProps }) => (
+      <Alert
+        key={key}
+        title={toastTitle(title, toastType(type))}
+        variant={toastType(type)}
+        timeout={sticky ? false : 8000}
```

Comment:
`sticky` means that until the user didn't click on the close button, the alert should be visible, in the older implementation we did:
```
const ToastComponent = sticky
        ? ToastNotification
        : TimedToastNotification;
```

where the `timed` implementation is what you use here,
maybe we can use the regular Alert component for sticky?

Reply:
Hmm I don't understand, is there a problem with PF4 components? For me both `sticky` & `timeout` cases works well

Reply:
sorry, my bad, the `onTimeout` on the next line and `sticky` got me confused,